app/src/lib/git/rebase.ts

```diff
@@ -82,6 +86,73 @@ export async function getRebaseContext(
 
   return null
 }
+
+/** Regex for identifying when rebase applied each commit onto the base branch */
+const rebaseApplyingRe = /^Applying: (.*)/
+
+/**
+ * A parser to read and emit rebase progress from Git `stdout`
+ */
+class GitRebaseParser {
+  private currentCommitCount = 0
+
+  public constructor(
+    startCount: number,
+    private readonly totalCommitCount: number
+  ) {
+    this.currentCommitCount = startCount
+  }
+
+  public parse(line: string): IRebaseProgress | null {
+    const match = rebaseApplyingRe.exec(line)
+    if (match == null || match.length !== 2) {
+      // Git will sometimes emit other output (for example, when it tries to
+      // resolve conflicts) and this does not match the expected output
+      return null
+    }
+
+    const commitSummary = match[1]
+    this.currentCommitCount++
+
+    const value = this.currentCommitCount / this.totalCommitCount
+
+    return {
+      kind: 'rebase',
+      title: `Rebasing ${this.currentCommitCount} of ${
+        this.totalCommitCount
+      } commits`,
+      value,
+      commitSummary,
+    }
+  }
+}
+
+function configureOptionsForRebase(
+  options: IGitExecutionOptions,
+  progress?: RebaseProgressOptions
+) {
+  if (!progress) {
```

Comment:
Just to be clear, I think the pattern you're using here is nice and unambiguous but it seems like we've converged on being explicit (https://github.com/desktop/desktop/issues/3538) so for consistency could we make this into a triple equals check please?

```suggestion
  if (progress === undefined) {
```